### ISSUE TITLE
Create a new "matrix-bridge" role to be reused with an existing installation

### ISSUE THREAD

**Issue by gnouts** [spantaleev/matrix-docker-ansible-deploy#174]
Hi all,

I already have my synapse installation (and don't want to change that part), but I'm interested in the bridge setup part...
So is it possible to use the bridges setup on my existing installation ? It's a classic debian install.

_I don't know Ansible, I'm a SaltStack guy :smile:_

**Comment by spantaleev** [comment]
You can disable lots of playbook functionality, but disabling the `matrix-synapse` role is not possible (yet), because it's a crucial component that everything else builds around.

Additionally, all bridges are part of the `matrix-synapse` role and hardcode configuration values which assume a `matrix-synapse` systemd service.

Thinking about it, those bridges don't necessarily need to live in the `matrix-synapse` role and are not dependent on Synapse itself, but rather on on the [Application Service API](https://matrix.org/docs/spec/application_service/r0.1.0.html).

It might make sense to extract the brdiges into a new role (`matrix-bridge`) or have each bridge be its own separate role. I guess we'll do that at some point.

For now, I'm not sure how easy and useful it would be to disable everything else and just leave bridge installation only.

**Comment by gnouts** [comment]
Thanks for the explanation. I'll keep an eye on this issue then :)

**Comment by spantaleev** [comment]
I've started working on this.

I decided that instead of a single `matrix-bridge` role (which would tend to be large and increase with time), it'd be nicer if we have a separate role for each bridge.

I've already done the work that migrates the mautrix-telegram and mautrix-whatsapp bridges into their own roles. The work-in-progress code is available in the `separate-bridge-roles` branch. As it develops, it will get merged to `master` via #179.

If it's any of these 2 bridges that you'd like to use, feel free to give it a try even now, before it gets merged.
